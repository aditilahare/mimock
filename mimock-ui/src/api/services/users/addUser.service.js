import { post } from 'api/AxiosClient';
import Bcrypt from 'bcryptjs';

export const addNewUser = async ({ name, userName, password, userRole }) => {
	const salt = Bcrypt.genSaltSync(12);
	const hashedPassword = Bcrypt.hashSync(password, salt);

	return await post(
		'/admin/users',
		{
			name,
			userName,
			userRole,
			password: hashedPassword,
		},
		undefined,
		{ showFullPageLoader: true }
	).catch((err) => Promise.reject(err));
};
