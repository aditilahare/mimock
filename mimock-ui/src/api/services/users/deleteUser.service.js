import { remove } from 'api/AxiosClient';

export const deleteUser = async (userName) => {
	const url = '/admin/users/delete-user?userName=' + userName;

	return await remove(url)
		.then((res) => res.data)
		.catch((err) => Promise.reject(err));
};
