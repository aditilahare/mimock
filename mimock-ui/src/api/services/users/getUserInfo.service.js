import { get } from 'api/AxiosClient';

export const getUserInfo = async (userId) => {
	const url = `/admin/users/user-info?userId=${userId}`;

	return await get(url, { showFullPageLoader: true })
		.then((res) => res.data.userInfo)
		.catch((err) => Promise.reject(err));
};
