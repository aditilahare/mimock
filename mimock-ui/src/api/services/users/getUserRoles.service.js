import { get } from 'api/AxiosClient';

export const getUserRoles = async () => {
	return await get('/static-records/user-roles')
		.then((res) => res.data)
		.catch((err) => err);
};
