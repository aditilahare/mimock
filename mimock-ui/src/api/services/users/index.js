import { getAllUsers } from './getUsers.service';
import { deleteUser } from './deleteUser.service';

export { getAllUsers, deleteUser };
