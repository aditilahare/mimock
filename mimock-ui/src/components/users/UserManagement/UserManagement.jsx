import React from 'react';
import { UserManagementWrapper } from './UserManagement.style';
import Header from './Header';
import UserTable from './UsersTable';

export default function UserManagement() {
	return (
		<UserManagementWrapper data-testid='user-management'>
			<Header />
			<UserTable />
		</UserManagementWrapper>
	);
}
