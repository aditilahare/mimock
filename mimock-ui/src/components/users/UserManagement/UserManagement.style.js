import tw from 'tailwind-styled-components';

export const UserManagementWrapper = tw.div`
  w-full
  h-full
`;
