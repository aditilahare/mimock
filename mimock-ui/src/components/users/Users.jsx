import React from 'react';
import UserManagement from './UserManagement';

export default function Users() {
	return <UserManagement />;
}
