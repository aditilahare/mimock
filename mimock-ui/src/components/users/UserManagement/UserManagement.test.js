import React from 'react';
import { render, act } from '@testing-library/react';
import { getAllUsers } from 'services/users';
import UserManagement from './UserManagement';

let mockedRecoilFn;
jest.mock('recoil', () => {
	mockedRecoilFn = jest.fn();
	return {
		atom: jest.fn(),
		useRecoilState: mockedRecoilFn.mockImplementation(() => [false, jest.fn()]),
	};
});
jest.mock('services/users');
jest.mock('react-router-dom');

describe('UserManagement', () => {
	beforeEach(() => {
		getAllUsers.mockResolvedValue([
			{
				userId: '1',
				userName: 'user1',
				name: 'User 1',
				userRole: 'ADMIN',
				isUserActive: true,
				userCreatedAt: '2022-04-28 19:42:21',
			},
		]);
	});

	it('should render user management', async () => {
		let tree;
		await act(async () => {
			tree = await render(<UserManagement />);
		});

		const { getByTestId, container } = tree;

		expect(getByTestId('user-management')).toBeInTheDocument();
		expect(getByTestId('user-management-header')).toBeInTheDocument();
		expect(getByTestId('user-table')).toBeInTheDocument();
		expect(getByTestId('add-user-btn')).toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});
});
