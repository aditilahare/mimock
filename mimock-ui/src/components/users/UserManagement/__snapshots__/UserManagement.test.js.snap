// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserManagement should render user management 1`] = `
<div>
  <div
    class="w-full h-full"
    data-testid="user-management"
  >
    <div
      class="flex justify-between items-center align-middle px-4 py-4 my-10 border-b-2 border-gray-100 w-11/12 mx-auto"
    >
      <div
        class="flex items-center justify-center align-middle font-sans text-2xl font-bold text-gray-700"
        data-testid="user-management-header"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 288c79.5 0 144-64.5 144-144S335.5 0 256 0 112 64.5 112 144s64.5 144 144 144zm128 32h-55.1c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16H128C57.3 320 0 377.3 0 448v16c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48v-16c0-70.7-57.3-128-128-128z"
          />
        </svg>
        <div
          class="mx-2 select-none"
        >
          User Management
        </div>
      </div>
      <button
        class="flex items-center justify-center gap-3 px-9 py-3 shadow-md rounded-md text-center text-lg font-semibold cursor-pointer hover:brightness-105 text-lime-700 dark:text-white w-fit bg-green-300 dark:bg-green-400"
        data-testid="add-user-btn"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15 2.013H9V9H2v6h7v6.987h6V15h7V9h-7z"
          />
        </svg>
        ADD USER
      </button>
    </div>
    <div
      class="w-full my-10 mx-auto"
      data-testid="user-table"
    >
      <div
        class="w-11/12 flex flex-row justify-between p-4 bg-gray-100 border border-gray-300 rounded-md shadow-sm mx-auto items-center align-middle my-1"
        data-testid="user-mgmt-table-header"
      >
        <div
          class="font-sans font-bold text-center antialiased text-gray-600 w-1/4"
          data-testid="header-item-name"
        >
          Name
        </div>
        <div
          class="font-sans font-bold text-center antialiased text-gray-600 w-1/4"
          data-testid="header-item-userName"
        >
          User name
        </div>
        <div
          class="font-sans font-bold text-center antialiased text-gray-600 w-1/4"
          data-testid="header-item-role"
        >
          Role
        </div>
        <div
          class="font-sans font-bold text-center antialiased text-gray-600 w-1/4"
          data-testid="header-item-activationStatus"
        >
          Activation status
        </div>
        <div
          class="font-sans font-bold text-center antialiased text-gray-600 w-1/4"
          data-testid="header-item-createdOn"
        >
          Created on
        </div>
        <div
          class="font-sans font-bold text-center antialiased text-gray-600 w-1/4"
          data-testid="header-item-actions"
        >
          Actions
        </div>
      </div>
      <div
        class="w-11/12 flex flex-row justify-between items-center align-middle p-4 bg-white border border-gray-200 rounded-md mx-auto my-1"
        data-testid="user-record-user1"
      >
        <div
          class="font-sans text-gray-800 text-center w-1/4"
          data-testid="user-name"
        >
          User 1
        </div>
        <div
          class="font-sans text-gray-800 text-center w-1/4"
          data-testid="user-userName"
        >
          user1
        </div>
        <div
          class="w-1/4"
          data-testid="user-role"
        >
          <div
            class="font-sans font-semibold text-white text-sm text-center antialiased mx-auto w-24 p-1 px-2 rounded-sm shadow-sm bg-teal-400"
            data-testid="role-pill-ADMIN"
            role="ADMIN"
          >
            ADMIN
          </div>
        </div>
        <div
          class="w-1/4 flex flex-row justify-between items-center align-middle gap-2"
        >
          <div
            class="flex flex-row justify-between items-center align-middle gap-2 mx-auto"
            data-testid="user-activation-status"
          >
            <div
              class="w-1"
              data-testid="status-indicator-true"
            >
              <div
                class="rounded-full mx-auto block p-2 text-center bg-green-500"
              />
            </div>
            <div
              class="w-24 font-sans font-semibold text-center text-gray-700"
              data-testid="status-label-true"
            >
              Active
            </div>
          </div>
        </div>
        <div
          class="w-1/4 font-sans antialiased text-center text-gray-600 text-xs font-semibold"
          data-testid="user-created-timestamp"
        >
          28 Apr 2022
        </div>
        <div
          class="w-1/4"
          data-testid="user-action-user1"
        >
          <div
            class="text-center text-2xl text-gray-600 cursor-pointer flex justify-center items-center align-middle gap-4 w-full"
          >
            <button
              aria-label="Edit User"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              data-mui-internal-clone-element="true"
              data-testid="edit-user1"
              tabindex="0"
              type="button"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 576 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="Delete User"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              data-mui-internal-clone-element="true"
              data-testid="delete-user1"
              tabindex="0"
              type="button"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
